### PR TITLE
Eliminate Logic.getKeyForStudent() #1136

### LINKED ISSUE #1136: Eliminate Logic.getKeyForStudent()
_From [dam...@gmail.com](https://code.google.com/u/114919422028123491955/) on August 16, 2013 20:31:19_

Instead of this, we should be able to use Logic.getEncryptedKeyForStudent()
The unencrypted key should not be visible outside Logic.

_Original issue: http://code.google.com/p/teammatespes/issues/detail?id=1137_

### PR DESCRIPTION
**Pull request by wkurniawan07**
Fixes #1136

### CODE REVIEW

**Comment by LowWeiLin** [comment]
Can we put this on hold? There will be significant conflicts with the issue i'm working on to migrate the student entity.

**Comment by wkurniawan07** [comment]
Not an issue at all. Curious though, what's the conflict going to be like?

File: src/main/java/teammates/storage/api/StudentsDb.java [hunk 1]
```diff
@@ -187,30 +187,17 @@ public StudentAttributes getStudentForGoogleId(String courseId, String googleId)
...
-                    KeyFactory.stringToKey(registrationKey));
-            studentAttributes = new StudentAttributes(student); 
+                    KeyFactory.stringToKey(originalKey));
+            return new StudentAttributes(student); 
         } catch (Exception e) {
-            try {
-                //Failing that, we try to retrieve assuming the given registrationKey is unencrypted 
-                //  (early versions of the system sent unencrypted keys).
-                //TODO: This branch can be removed after Dec 2013
-                Student student = getPM().getObjectById(Student.class,
```
Comment (thyageshm) [inline src/main/java/teammates/storage/api/StudentsDb.java]: @damithc just fyi

Comment (damithc) [inline src/main/java/teammates/storage/api/StudentsDb.java]: OK to remove.

**Comment by thyageshm** [comment]
@wkurniawan07 any reason why this pr is on hold?

**Comment by wkurniawan07** [comment]
@LowWeiLin I'm actually very curious now; how is it going to conflict?

**Comment by wkurniawan07** [comment]
So this PR was put on hold for nothing 😑😑
(and I honestly haven't seen how it's going to conflict with EC issue)

**Comment by kanghj** [comment]
At some point, the intended solution was to change the schema, making the current way of handling keys obsolete. Not sure if that was done in the end, but that might have made this PR obsolete too, hence the conflict. I won't be getting to that issue anytime in the next week or so though.

File: src/test/java/teammates/test/cases/storage/StudentsDbTest.java [hunk 2]
```diff
@@ -135,9 +135,8 @@ public void testGetStudent() throws InvalidParametersException, EntityDoesNotExi
         ______TS("typical success case: existent");
         StudentAttributes retrieved = studentsDb.getStudentForEmail(s.course, s.email);
         assertNotNull(retrieved);
-        assertNotNull(studentsDb.getStudentForRegistrationKey(retrieved.key));
```
Comment (kanghj) [inline src/test/java/teammates/test/cases/storage/StudentsDbTest.java]: Hmm, why is this removed?

Comment (kanghj) [inline src/test/java/teammates/test/cases/storage/StudentsDbTest.java]: Oh, wait, misread this.

File: src/main/java/teammates/ui/controller/FeedbackSubmissionEditPageData.java [hunk 3]
```diff
@@ -195,6 +195,10 @@ public void setSubmitAction(String submitAction) {
         return result;
     }
     
+    public String getEncryptedRegkey() {
+        return StringHelper.encrypt(student.key);
+    }
```
Comment (kanghj) [inline src/main/java/teammates/ui/controller/FeedbackSubmissionEditPageData.java]: Not exactly related to this issue, but should we have a new field `encryptedKey` on StudentAttributes (although not persisted to the database)? Other than the search document's id, it doesn't actually seem like there is a valid use of `student.key` anywhere.

File: src/main/java/teammates/logic/api/Logic.java [hunk 4]
```diff
@@ -439,13 +439,13 @@ public InstructorAttributes getInstructorForRegistrationKey(String encryptedKey)
      * @return null if the key doesn't exist.
      * @throws EntityDoesNotExistException
      */
-    public String getKeyForInstructor(String courseId, String email)
+    public String getEncryptedKeyForInstructor(String courseId, String email)
```
Comment (kanghj) [inline src/main/java/teammates/logic/api/Logic.java]: Change the javadoc for this, it says `Get the decrypted registration key` currently.

Comment (kanghj) [inline src/main/java/teammates/logic/api/Logic.java]: `@return null if the key doesn't exist` is incorrect too.

File: src/main/java/teammates/ui/controller/FeedbackSubmissionEditPageData.java (hunk 3 above)
Comment (wkurniawan07) [inline src/main/java/teammates/ui/controller/FeedbackSubmissionEditPageData.java]: Good thing to consider. For `InstructorAttributes` as well.

**Comment by wkurniawan07** [comment]
@kanghj changes made
